config/src/main/java/com/yahoo/config/subscription/impl/ConfigSubscription.java

```diff
@@ -129,50 +159,52 @@ public boolean equals(Object o) {
         return false;
     }
 
-    void setConfigChanged(boolean changed) {
-        this.configChanged = changed;
-    }
-
-    void setGenerationChanged(boolean genChanged) {
-        this.generationChanged = genChanged;
-    }
 
     /**
      * Called from {@link ConfigSubscriber} when the changed status of this config is propagated to the clients
      */
-    public void resetChangedFlags() {
-        setConfigChanged(false);
-        setGenerationChanged(false);
+    public boolean isConfigChangedAndReset() {
```

Comment:
this method should limit the try-clear loop to the current generation

Reply:
Fixed